### DISCUSSION TITLE
Registry to work with Kafka SASL PLAIN

### DISCUSSION THREAD

**Discussion by spakendralo** [Apicurio/apicurio-registry#2475] [General]
Hello, 

I would like to know if APICurio Schema Registry could work with SASL PLAIN, as I already have a Kafka installation and it uses PLAIN and not OAuth. How could that be integrated?

**Comment by carlesarnal** [comment]
No, the only supported options are SCRAM and OAUTH. It should be possible to make it work with PLAIN, but you'll need to modify the application and build your own image.

**Reply by eldengates** [reply]
Hi, @carlesarnal my kafka cluster is configured
SCRAM_SHA-512 for authentication
SASL_PLAINTEXT for encryption
can you please provide which env should i export in this example? to make it work. I tried like below but didn't worked. I can provide additional info by request.
- env:
        - name: KAFKA_SECURITY_PROTOCOL
          value: SASL_PLAINTEXT
        - name: KAFKA_SASL_MECHANISM
          value: SCRAM
        - name: KAFKA_SASL_JAAS_CONFIG
          value: org.apache.kafka.common.security.scram.ScramLoginModule required username="user1" password="pw1";
